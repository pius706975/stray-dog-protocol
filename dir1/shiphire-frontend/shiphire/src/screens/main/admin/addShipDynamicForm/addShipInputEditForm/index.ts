export { default as AddShipInputEditForm } from './AddShipInputEditForm';
