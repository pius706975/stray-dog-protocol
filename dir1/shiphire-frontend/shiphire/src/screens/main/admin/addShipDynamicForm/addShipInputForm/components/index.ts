export { default as InputPreview } from './InputPreview';
export { default as SelectInputProperties } from './SelectInputProperties';
export { default as AdditionalInput } from './AdditionalInput';
