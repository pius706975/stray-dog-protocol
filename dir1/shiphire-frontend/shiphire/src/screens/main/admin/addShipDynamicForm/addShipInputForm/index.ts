export { default as AddShipInputForm } from './AddShipInputForm';
