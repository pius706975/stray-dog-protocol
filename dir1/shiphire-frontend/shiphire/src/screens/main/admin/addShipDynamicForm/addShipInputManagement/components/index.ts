export { default as DetailInputSheet } from './DetailInputSheet';
export { default as ModalConfirmation } from './ModalConfirmation';
