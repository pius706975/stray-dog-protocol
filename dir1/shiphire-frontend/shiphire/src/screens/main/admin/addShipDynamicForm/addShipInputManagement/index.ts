export { default as AddShipInputManagement } from './AddShipInputManagement';
