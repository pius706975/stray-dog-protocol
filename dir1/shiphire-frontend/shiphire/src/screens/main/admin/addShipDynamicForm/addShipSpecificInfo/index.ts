export { default as AddShipSpecificInfo } from './AddShipSpecificInfo';
