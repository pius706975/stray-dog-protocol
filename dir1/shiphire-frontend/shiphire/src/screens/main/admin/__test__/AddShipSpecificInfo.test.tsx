import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MainAdminStackParamList } from '../../../../types';
import httpRequest from '../../../../services/api';
import MockAdapter from 'axios-mock-adapter';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';
import {
    useDeleteShipSpectAddShip,
    useGetAddShipDynamicInputByTemplateName,
    useGetShipSpecification,
} from '../../../../hooks';
import { AddShipSpecificInfo } from '../addShipDynamicForm/addShipSpecificInfo';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { AddShipInputForm } from '../addShipDynamicForm/addShipInputForm';
import { AddShipInputManagement } from '../addShipDynamicForm/addShipInputManagement';

jest.mock('../../../../hooks');

const queryClient = new QueryClient();
const MainAdminStack = createNativeStackNavigator<MainAdminStackParamList>();

const mock = new MockAdapter(httpRequest);

const adminStackMockComponent = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainAdminStack.Navigator>
                    <MainAdminStack.Screen
                        name="AddShipSpecificInfo"
                        component={AddShipSpecificInfo}
                    />
                    <MainAdminStack.Screen
                        name="AddShipInputForm"
                        component={AddShipInputForm}
                        initialParams={{ templateType: '' }}
                    />
                    <MainAdminStack.Screen
                        name="AddShipInputManagement"
                        component={AddShipInputManagement}
                        initialParams={{ templateType: 'bargeSpesific' }}
                    />
                </MainAdminStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

const mockShipSpect = [
    { templateType: 'bargeSpesific', value: 'barge spesific' },
    { templateType: 'ferrySpesific', value: 'ferry spesific' },
];
beforeEach(() => {
    mock.onGet('/admin/get-dynamic-input-spec').reply(200, {
        message: 'success',
        data: mockShipSpect,
    });
    (useGetShipSpecification as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onSuccess } = options;
            const mockResponseData = {
                data: {
                    data: mockShipSpect,
                },
            };
            onSuccess(mockResponseData);
        }),
    }));
});

describe('Testing add ship spesific info test', () => {
    describe('Snapshot testing', () => {
        it('should render add ship spesific info test screen correctly', () => {
            const specificInfoScreen = render(adminStackMockComponent);
            expect(specificInfoScreen).toMatchSnapshot();
        });
    });
    describe('Unit testing', () => {
        it('should navigate to add specific screen when press + icon', () => {
            const { getByTestId } = render(adminStackMockComponent);
            const addSpecificBtn = getByTestId('add-specific-btn');
            fireEvent.press(addSpecificBtn);

            expect(getByTestId('AddShipInputForm')).toBeDefined();
        });
        it('should navigate to input management screen when press item', () => {
            mock.onGet('/admin/get-dynamic-input').reply(200, {
                message: 'success',
                data: [],
            });
            (
                useGetAddShipDynamicInputByTemplateName as jest.Mock
            ).mockImplementation(() => ({
                mutate: jest.fn().mockImplementation((data, options) => {
                    const { onSuccess } = options;
                    const mockResponseData = {
                        data: {
                            data: [],
                        },
                    };
                    onSuccess(mockResponseData);
                }),
            }));

            const { getByTestId } = render(adminStackMockComponent);
            const inputManageBtn = getByTestId('input-manage-0');
            fireEvent.press(inputManageBtn);

            expect(getByTestId('addshipinputmanagement')).toBeDefined();
        });
        it('should delete specific item succesfully', async () => {
            mock.onPost('/admin/delete-specific').reply(200, {
                message: 'success',
            });
            (useDeleteShipSpectAddShip as jest.Mock).mockImplementation(() => ({
                mutate: jest.fn().mockImplementation((data, options) => {
                    const { onSuccess } = options;
                    const mockResponseData = {
                        data: {
                            message: 'success',
                        },
                    };
                    onSuccess(mockResponseData);
                }),
            }));

            const { getByTestId } = render(adminStackMockComponent);
            const deleteBtn = getByTestId('btn-delete-1');
            fireEvent.press(deleteBtn);
            fireEvent.press(getByTestId('confirm-delete'));

            await waitFor(() => {
                const modalState = store.getState().modal;
                expect(modalState).toEqual({
                    visible: true,
                    status: 'success',
                    text: 'Delete successfully',
                });
            });
        });
        it('should show modal when error deleting', async () => {
            mock.onPost('/admin/delete-specific').reply(500, {
                message: 'failed',
            });
            (useDeleteShipSpectAddShip as jest.Mock).mockImplementation(() => ({
                mutate: jest.fn().mockImplementation((data, options) => {
                    const { onError } = options;
                    const mockResp = {
                        data: { message: 'failed' },
                    };
                    onError(mockResp);
                }),
            }));

            const { getByTestId } = render(adminStackMockComponent);
            const deleteBtn = getByTestId('btn-delete-1');
            fireEvent.press(deleteBtn);
            fireEvent.press(getByTestId('confirm-delete'));

            await waitFor(() => {
                const modalState = store.getState().modal;
                expect(modalState).toEqual({
                    visible: true,
                    status: 'failed',
                    text: 'Cannot delate specification',
                });
            });
        });
    });
});
