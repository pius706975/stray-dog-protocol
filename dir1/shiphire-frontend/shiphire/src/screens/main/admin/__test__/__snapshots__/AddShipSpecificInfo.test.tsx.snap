// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing add ship spesific info test Snapshot testing should render add ship spesific info test screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="AddShipSpecificInfo"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      >
        <RNSScreenStackHeaderSubview
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          type="right"
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="add-specific-btn"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="24"
              bbWidth="24"
              fill="none"
              focusable={false}
              height={24}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "flex": 0,
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
              width={24}
              xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g clip-path="url(#clip0_77_28470)">
    <path d="M18 13H13V18C13 18.55 12.55 19 12 19C11.45 19 11 18.55 11 18V13H6C5.45 13 5 12.55 5 12C5 11.45 5.45 11 6 11H11V6C11 5.45 11.45 5 12 5C12.55 5 13 5.45 13 6V11H18C18.55 11 19 11.45 19 12C19 12.55 18.55 13 18 13Z" fill="#323232"/>
    </g>
    <defs>
    <clipPath id="clip0_77_28470">
    <rect width="24" height="24" fill="white"/>
    </clipPath>
    </defs>
    </svg>
    
    
    "
              xmlns="http://www.w3.org/2000/svg"
            >
              <RNSVGGroup
                fill={null}
                propList={
                  [
                    "fill",
                  ]
                }
              >
                <RNSVGGroup
                  clipPath="clip0_77_28470"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGPath
                    d="M18 13H13V18C13 18.55 12.55 19 12 19C11.45 19 11 18.55 11 18V13H6C5.45 13 5 12.55 5 12C5 11.45 5.45 11 6 11H11V6C11 5.45 11.45 5 12 5C12.55 5 13 5.45 13 6V11H18C18.55 11 19 11.45 19 12C19 12.55 18.55 13 18 13Z"
                    fill={
                      {
                        "payload": 4281479730,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
                <RNSVGDefs>
                  <RNSVGClipPath
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    name="clip0_77_28470"
                  >
                    <RNSVGRect
                      fill={
                        {
                          "payload": 4294967295,
                          "type": 0,
                        }
                      }
                      height="24"
                      propList={
                        [
                          "fill",
                        ]
                      }
                      width="24"
                      x="0"
                      y="0"
                    />
                  </RNSVGClipPath>
                </RNSVGDefs>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </RNSScreenStackHeaderSubview>
      </RNSScreenStackHeaderConfig>
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTScrollView
            data={
              [
                "",
              ]
            }
            getItem={[Function]}
            getItemCount={[Function]}
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            showsVerticalScrollIndicator={false}
            stickyHeaderIndices={[]}
            style={
              {
                "height": "100%",
                "width": "100%",
              }
            }
            testID="addShipSpecInfo"
            viewabilityConfigCallbackPairs={[]}
          >
            <View>
              <View
                onFocusCapture={[Function]}
                onLayout={[Function]}
                style={null}
              >
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "borderRadius": 8,
                        "margin": 6,
                      },
                    ]
                  }
                >
                  <View
                    ItemSeparatorComponent={[Function]}
                    data={
                      [
                        {
                          "templateType": "bargeSpesific",
                          "value": "barge spesific",
                        },
                        {
                          "templateType": "ferrySpesific",
                          "value": "ferry spesific",
                        },
                      ]
                    }
                    getItem={[Function]}
                    getItemCount={[Function]}
                    keyExtractor={[Function]}
                    onContentSizeChange={[Function]}
                    onEndReachedThreshold={0.1}
                    onLayout={[Function]}
                    onMomentumScrollBegin={[Function]}
                    onMomentumScrollEnd={[Function]}
                    onScroll={[Function]}
                    onScrollBeginDrag={[Function]}
                    onScrollEndDrag={[Function]}
                    removeClippedSubviews={false}
                    renderItem={[Function]}
                    scrollEventThrottle={50}
                    stickyHeaderIndices={[]}
                    viewabilityConfigCallbackPairs={[]}
                  >
                    <View
                      onFocusCapture={[Function]}
                      onLayout={[Function]}
                      style={null}
                    >
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "alignItems": "center",
                              "borderBottomWidth": 0.5,
                              "flexDirection": "row",
                              "margin": 5,
                              "padding": 10,
                            },
                          ]
                        }
                      >
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "width": "90%",
                            }
                          }
                          testID="input-manage-0"
                        >
                          <View
                            style={
                              [
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                {
                                  "flexDirection": "row",
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                [
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  {
                                    "justifyContent": "flex-start",
                                    "padding": 5,
                                    "paddingLeft": 0,
                                    "paddingRight": 10,
                                  },
                                ]
                              }
                            />
                            <Text
                              numberOfLines={0}
                              style={
                                {
                                  "borderBottomWidth": 0,
                                  "borderColor": undefined,
                                  "color": "#4d5963",
                                  "fontFamily": "Kanit-Regular",
                                  "fontSize": 18,
                                  "lineHeight": undefined,
                                  "textAlign": undefined,
                                }
                              }
                            >
                              barge spesific
                            </Text>
                          </View>
                        </View>
                        <View
                          accessibilityRole="button"
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": false,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onLayout={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "alignItems": "center",
                              "backgroundColor": "#bcd3e6",
                              "borderRadius": 999,
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "minWidth": undefined,
                              "opacity": 1,
                              "padding": 4,
                              "paddingHorizontal": undefined,
                              "paddingVertical": undefined,
                            }
                          }
                          testID="btn-delete-0"
                        >
                          <RNSVGSvgView
                            align="xMidYMid"
                            bbHeight="28"
                            bbWidth="28"
                            focusable={false}
                            height={28}
                            meetOrSlice={0}
                            minX={0}
                            minY={-960}
                            style={
                              [
                                {
                                  "backgroundColor": "transparent",
                                  "borderWidth": 0,
                                },
                                {
                                  "flex": 0,
                                  "height": 28,
                                  "width": 28,
                                },
                              ]
                            }
                            vbHeight={960}
                            vbWidth={960}
                            width={28}
                            xml="<svg xmlns="http://www.w3.org/2000/svg" height="28" viewBox="0 -960 960 960" width="28"><path d="M312-144q-29.7 0-50.85-21.15Q240-186.3 240-216v-480h-48v-72h192v-48h192v48h192v72h-48v479.566Q720-186 698.85-165 677.7-144 648-144H312Zm336-552H312v480h336v-480ZM384-288h72v-336h-72v336Zm120 0h72v-336h-72v336ZM312-696v480-480Z" fill="#CC1F12"/></svg>
         "
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <RNSVGGroup
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                            >
                              <RNSVGPath
                                d="M312-144q-29.7 0-50.85-21.15Q240-186.3 240-216v-480h-48v-72h192v-48h192v48h192v72h-48v479.566Q720-186 698.85-165 677.7-144 648-144H312Zm336-552H312v480h336v-480ZM384-288h72v-336h-72v336Zm120 0h72v-336h-72v336ZM312-696v480-480Z"
                                fill={
                                  {
                                    "payload": 4291567378,
                                    "type": 0,
                                  }
                                }
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                              />
                            </RNSVGGroup>
                          </RNSVGSvgView>
                        </View>
                      </View>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                          ]
                        }
                      />
                    </View>
                    <View
                      onFocusCapture={[Function]}
                      onLayout={[Function]}
                      style={null}
                    >
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "alignItems": "center",
                              "borderBottomWidth": 0.5,
                              "flexDirection": "row",
                              "margin": 5,
                              "padding": 10,
                            },
                          ]
                        }
                      >
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "width": "90%",
                            }
                          }
                          testID="input-manage-1"
                        >
                          <View
                            style={
                              [
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                {
                                  "flexDirection": "row",
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                [
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  {
                                    "justifyContent": "flex-start",
                                    "padding": 5,
                                    "paddingLeft": 0,
                                    "paddingRight": 10,
                                  },
                                ]
                              }
                            />
                            <Text
                              numberOfLines={0}
                              style={
                                {
                                  "borderBottomWidth": 0,
                                  "borderColor": undefined,
                                  "color": "#4d5963",
                                  "fontFamily": "Kanit-Regular",
                                  "fontSize": 18,
                                  "lineHeight": undefined,
                                  "textAlign": undefined,
                                }
                              }
                            >
                              ferry spesific
                            </Text>
                          </View>
                        </View>
                        <View
                          accessibilityRole="button"
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": false,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onLayout={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "alignItems": "center",
                              "backgroundColor": "#bcd3e6",
                              "borderRadius": 999,
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "minWidth": undefined,
                              "opacity": 1,
                              "padding": 4,
                              "paddingHorizontal": undefined,
                              "paddingVertical": undefined,
                            }
                          }
                          testID="btn-delete-1"
                        >
                          <RNSVGSvgView
                            align="xMidYMid"
                            bbHeight="28"
                            bbWidth="28"
                            focusable={false}
                            height={28}
                            meetOrSlice={0}
                            minX={0}
                            minY={-960}
                            style={
                              [
                                {
                                  "backgroundColor": "transparent",
                                  "borderWidth": 0,
                                },
                                {
                                  "flex": 0,
                                  "height": 28,
                                  "width": 28,
                                },
                              ]
                            }
                            vbHeight={960}
                            vbWidth={960}
                            width={28}
                            xml="<svg xmlns="http://www.w3.org/2000/svg" height="28" viewBox="0 -960 960 960" width="28"><path d="M312-144q-29.7 0-50.85-21.15Q240-186.3 240-216v-480h-48v-72h192v-48h192v48h192v72h-48v479.566Q720-186 698.85-165 677.7-144 648-144H312Zm336-552H312v480h336v-480ZM384-288h72v-336h-72v336Zm120 0h72v-336h-72v336ZM312-696v480-480Z" fill="#CC1F12"/></svg>
         "
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <RNSVGGroup
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                            >
                              <RNSVGPath
                                d="M312-144q-29.7 0-50.85-21.15Q240-186.3 240-216v-480h-48v-72h192v-48h192v48h192v72h-48v479.566Q720-186 698.85-165 677.7-144 648-144H312Zm336-552H312v480h336v-480ZM384-288h72v-336h-72v336Zm120 0h72v-336h-72v336ZM312-696v480-480Z"
                                fill={
                                  {
                                    "payload": 4291567378,
                                    "type": 0,
                                  }
                                }
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                              />
                            </RNSVGGroup>
                          </RNSVGSvgView>
                        </View>
                      </View>
                    </View>
                  </View>
                  <Modal
                    animationType="slide"
                    hardwareAccelerated={false}
                    onRequestClose={[Function]}
                    transparent={true}
                    visible={false}
                  />
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
